src/test_report.py

```diff
@@ -46,16 +46,21 @@ def _get_group_stats(self, parent_id):
             })
         }
```

Comment:
I am unsure about group stats here. Should I use a total number of distinct sets of nodes (groups) as `total` or keep it as is with the number of immediate child nodes of the `checkout` node?
I think we need to use the first approach to make the stats consistent with the list of test results.

Reply:
Yes, this is a bit unclear.  Also I think the name `group` is misleading.  These are really `jobs` in the same sense as when using `kci job` to submit something that will produce a set of results.  What mattered in the legacy system was the total number of test results rather than jobs, so counting how many nodes have a result set would give us this information.

Reply:
Makes sense. Updated code to use `job` rather than `group`.
Also, update logic to get statistics. It will denote the number of distinct job runs and failures rather than child nodes of `checkout` node.